src/renderer/contexts/ExecutionContext.tsx

```diff
@@ -209,6 +209,16 @@ export const ExecutionProvider = memo(({ children }: React.PropsWithChildren<{}>
 
     const [isBackendKilled, setIsBackendKilled] = useState(false);
 
+    const [percentComplete, setPercentComplete] = useState<number | undefined>(undefined);
+
+    useEffect(() => {
+        if (percentComplete === undefined) {
+            ipcRenderer.send('set-progress-bar', -1);
```

Comment:
You spread the fact that -1 is interpreted as "none" all the way into the frontend. Could you change the API of `set-progress-bar` to take a `number | null`? This would make for a cleaner API and the handler can be implemented as a nice oneliner: `mainWindow.setProgressBar(progress ?? -1)`.